onnxruntime/core/providers/mkldnn/math/sum.cc

```diff
@@ -0,0 +1,224 @@
+// Copyright(C) 2018 Intel Corporation
+// Licensed under the MIT License
+
+#ifdef _WIN32
+#pragma warning(disable : 4244)
+#endif
+
+#include "core/providers/mkldnn/mkldnn_common.h"
+#include "core/providers/mkldnn/math/sum.h"
+#include "core/providers/mkldnn/mkldnn_fwd.h"
+
+namespace onnxruntime {
+namespace mkl_dnn {
+
+namespace {
+// Struct which encapsulates parameters for MKLDNN Sum primitives.
+struct SumParams {
+  std::vector<mkldnn::memory::dims>& src_dims;
+  mkldnn::memory::dims& dst_dim;
+  int num_inputs;
+  int num_dimensions;
+
+  SumParams(std::vector<mkldnn::memory::dims>& dims,
```

Comment:
can these be passed by const-ref?

Reply:
@pranavsharma src_dims can be const ref but we want dst_dims just refs. dst_dims will be modified after compute.